zerver/actions/users.py

```diff
@@ -138,6 +145,13 @@ def do_deactivate_user(
         change_user_is_active(user_profile, False)
 
         clear_scheduled_emails(user_profile.id)
+
+        if spammer and not user_profile.full_name.endswith(" (spammer)"):
+            check_change_full_name(user_profile, user_profile.full_name + " (spammer)", acting_user)
+
+        if message_delete_action is not None:
+            delete_deactivated_user_messages(user_profile, message_delete_action)
```

Comment:
Let's add a field to the `extra_data` dict in the `RealmAuditLog` entry below with the spammer/deletion parameters that were used for the deactivation.